src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -250,19 +249,25 @@ public static async Task<PackageSpec> GetProjectSpec(IDependencyGraphProject pro
 
                         if (!string.IsNullOrEmpty(outputPath))
                         {
-                            var persistedDGSpecPath = Path.Combine(outputPath, dgFileName);
-
-                            if (File.Exists(persistedDGSpecPath))
+                            for (int frameworkCount = 0; frameworkCount < packageSpec.RestoreMetadata.TargetFrameworks.Count; frameworkCount++)
```

Comment:
Previously the TargetFrameworks were not considered, why they were not before and are now? Will this influence any current scenario?

Reply:
They were consider previously as well. 

GetClosure walks all the project references, https://github.com/NuGet/NuGet.Client/blob/e0128993c9ade2eb98d0efec7e9808fec6980e44/src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs#L184. 

Specifically before a similar thing was being done, but the read from disk was happening to early when unnecessary.